sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/ir/expression/DBSPOpcode.java

```diff
@@ -19,6 +19,7 @@ public enum DBSPOpcode {
 
     // Binary operations
     ADD("+", false),
+    SAFE_ADD("safe_add", false),
```

Comment:
you also need to add this function to RustSqlRuntimeLibrary in the arithmeticFunctions map.
Then the casts will be properly inserted and I expect the test with casts will pass.